src/index.js

```diff
@@ -54,6 +54,15 @@ function map(array, callback) {
   return result;
 };
 
+function reduce(array, callback, initial) {
+  let accumulator = initial ?? array[0];
+  const startIndex = initial ? 0 : 1;
+  for (let i = startIndex; i < array.length; ++i) {
+    accumulator = callback(accumulator, array[i], i, array);
+  }
+  return accumulator;
+};
```

Comment:
This is looking good, but there's something not quite right 🤔 . Compare the following invocation outputs:
```js
robDuce(data, sumOfValues, 0) // outputs 9

data.reduce(sumOfValues, 0) // outputs 10
```

Reply:
robDuce 👌